src/perform_step/symplectic_perform_step.jl

```diff
@@ -171,7 +171,7 @@ end
   duprev, uprev = load_symp_state(integrator)
 
   # x(t+Δt) = x(t) + v(t)*Δt + 1/2*a(t)*Δt^2
-  ku = f.f1(duprev,uprev,p,t)
+  ku = integrator.fsallast.x[1]
```

Comment:
I feel this should be `integrator.fsallast.x[2]`

Reply:
That leads to immediate divergence. I'm not sure I fully understand the variable naming scheme but in that function it seems `ku` is `a(t)` and `kdu` is `a(t+dt)`. So once you reach the next step, you want to load `kdu` from the last step into the current `ku`.

Reply:
You are exactly, right.